### DISCUSSION TITLE
Client doesn’t resolve a hostname when pinging a hostname

### DISCUSSION THREAD

**Discussion by dingdang2012** [openthread/openthread#8664] [Q&A] [answered]
When pinging a hostname, Error 6: Parse is observed.
Build Steps:
a) git clone https://github.com/openthread/ot-nrf528xx.git --recursive
b) cd ot-nrf528xx
c) ./script/bootstrap
b) script/build nrf52840 USB_trans -DOT_SRP_CLIENT=ON -DOT_ECDSA=ON
c) arm-none-eabi-objcopy -O ihex build/bin/ot-cli-ftd ot-cli-ftd.hex
d) nrfjprog -f nrf52 --chiperase --program ot-cli-ftd.hex --reset
e)
GitHubGitHub
[GitHub - openthread/ot-nrf528xx: OpenThread on Nordic nRF528xx examples.](https://github.com/openthread/ot-nrf528xx.git)
OpenThread on Nordic nRF528xx examples. Contribute to openthread/ot-nrf528xx development by creating an account on GitHub. (53 kB)

**Comment by jwhui** [comment]
The `ping` CLI command only accepts IP addresses for the destination, and not a host name.

See the [ping CLI documentation](https://github.com/openthread/openthread/blob/main/src/cli/README.md#ping-async--i-source-ipaddr-size-count-interval-hoplimit-timeout) for more details.

**Reply by dingdang2012** [reply]
Is there a way to ping or testing connection to a host by using hostname?

**Reply by jwhui** [reply, marked as answer]
Use the [`dns resolve`](https://github.com/openthread/openthread/tree/main/src/cli#dns-resolve-hostname-dns-server-ip-dns-server-port-response-timeout-ms-max-tx-attempts-recursion-desired-boolean) command to resolve hostname to IP address.

**Reply by dingdang2012** [reply]
Thanks!
